src/plugins/workspace/server/routes/duplicate.ts

```diff
@@ -72,6 +74,8 @@ export const registerDuplicateRoute = (
         });
       }
 
+      const assignedDataSources = await getDataSourcesList(savedObjectsClient, [targetWorkspace]);
```

Comment:
What about we doing this in `importSavedObjectsFromStream`? I think the validation should apply to import as well.

Reply:
The method of `getDataSourcesList ` is in the workspace file, so I prefer to use the method in the `duplicate.ts`

Reply:
The code has updated, and removed `getDataSourcesList` from `src/plugins/workspace/server/routes/duplicate.ts`